### ISSUE TITLE
DccChatFactory fails to return protocol

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3179]
|<img alt="ldlework's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ldlework reported|
|-|-|
|Trac ID|trac#3179|
|Type|defect|
|Created|2008-04-11 23:14:33Z|
|Branch|https://github.com/twisted/twisted/tree/dccchatfactory-protocol-3179|

Line 1718 of /twisted/words/protocols/irc.py should be :

```
        return p
```

Attachments:

* [irc.py.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7dd/7dd058dd320cbccdf196f7874c1c86c9428e3141/8f5aa514a1890a453329861e2847c05bedd2a3fa.diff) (480 bytes) - added by Nagyman on 2012-03-13 20:23:36Z - 
* [test_irc.py.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7dd/7dd058dd320cbccdf196f7874c1c86c9428e3141/a365e6139c9470bb1852b99d553b3abe0b5ee330.diff) (979 bytes) - added by Nagyman on 2012-03-13 20:23:46Z - 
* [test_irc.py.2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7dd/7dd058dd320cbccdf196f7874c1c86c9428e3141/5bd9c9523ed2337cb3e2a75f6b7b2b309e0eef23.diff) (880 bytes) - added by Nagyman on 2012-03-13 20:32:23Z - Whitespace fixed. Ignore previous diff.
* [3179.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/7dd/7dd058dd320cbccdf196f7874c1c86c9428e3141/2954acdf5b6732bcd1cdc069fcf5bccb4cde0216.diff) (1616 bytes) - added by Nagyman on 2012-03-14 00:48:00Z - Diff for all changes, now includes a news file.

<details><summary>Searchable metadata</summary>

```
trac-id__3179 3179
type__defect defect
reporter__ldlework ldlework
priority__normal normal
milestone__ 
branch__branches_dccchatfactory_protocol_3179 branches/dccchatfactory-protocol-3179
branch_author__cyli__ldlework cyli, ldlework
status__closed closed
resolution__fixed fixed
component__words words
keywords__words_irc_dcc_chat_easy words irc dcc chat easy
time__1207955673000000 1207955673000000
changetime__1331728520000000 1331728520000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Nagyman's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Nagyman set owner to Nagyman<br>Nagyman set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Nagyman's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|Nagyman removed owner<br>Nagyman set status to `new`|
|-|-|

Sorry for the accidental whitespace in the first test_irc.py.diff.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

Thank you so much for the work, Nagyman!

Can you please include a [news file](http://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles)? 

It's basically a short description of what change has been made in this ticket.  That way when twisted is released, all these descriptions can be aggregated into the release notes.

That link has the information about description style, as well as where to put the file and what to name the file, etc.

Thanks again!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

Sorry, I can't seem to reassign this ticket back to you.  :(

**Comment by twisted-trac** [comment]
|<img alt="Nagyman's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_7"></a>|Nagyman commented|
|-|-|

New full patch including the required new file. Sorry!  I'm not sure why I don't show up in the users list for assignment...I seem to be able to accept ownership myself.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

(In [33783]) Branching to 'dccchatfactory-protocol-3179'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli commented|
|-|-|

(In [33785]) Apply patch from Nagyman

refs [#3179](https://github.com/twisted/twisted/issues/3179)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli commented|
|-|-|

This looks great, Nagyman!  Thanks so much for all your hard work.

The [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/dccchatfactory-protocol-3179) look good.  I'm going to merge.

One small nitpick I have for future patches is that there should be 3 lines between each class and the previous class (see the [Twisted coding standard](http://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html#auto5), the section on whitespace).

Once again, thanks!

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_11"></a>|@cyli set status to `closed`|
|-|-|

(In [33787]) Merge dccchatfactory-protocol-3179: Return protocol in t.w.protocols.irc.DccChatFactory.buildProtocol

Author: Nagyman
Reviewer: cyli
Fixes: [#3179](https://github.com/twisted/twisted/issues/3179)

Fix bug where the protocol was not returned from twisted.words.protocols.irc.DccChatFactory.buildProtocol

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

Here's some extra feedback about this change:

  1. `buildProtocol` should most likely have a docstring
  1. Since `buildProtocol` doesn't use the address passed to it, `None` might be a better dummy value than `"127.0.0.1"`.
  1. trial provides an `assertIsInstance` method which provides better information in the failure case than `assertTrue(isinstance(...))`.

Don't worry about this points now, just keep them in mind for future tickets.  Thanks again for your work on this.
